File: ConexionBD/ProveedorBD.cs
using punto_de_venta_C_.Modelo;
using MySqlConnector;
using System;
using System.Collections.Generic;

namespace punto_de_venta_C_.ConexionBD
{
    public class ProveedorBD
    {
        public List<Proveedor> obtenerTodos()
        {
            List<Proveedor> lista = new List<Proveedor>();
            string sql = "SELECT * FROM proveedores WHERE activo = 1 ORDER BY id";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand st = new MySqlCommand(sql, c))
                {
                    using (MySqlDataReader rs = st.ExecuteReader())
                    {
                        while (rs.Read())
                            lista.Add(mapear(rs));
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return lista;
        }

        public Proveedor buscarPorId(int id)
        {
            string sql = "SELECT * FROM proveedores WHERE id = @id";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@id", id);
                    using (MySqlDataReader rs = ps.ExecuteReader())
                    {
                        if (rs.Read())
                            return mapear(rs);
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return null;
        }

        public bool insertar(Proveedor p)
        {
            string sql = "INSERT INTO proveedores (nombre, contacto, telefono, email, direccion, activo) VALUES (@nombre,@contacto,@telefono,@email,@direccion,@activo)";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@nombre", p.getNombre());
                    ps.Parameters.AddWithValue("@contacto", p.getContacto());
                    ps.Parameters.AddWithValue("@telefono", p.getTelefono());
                    ps.Parameters.AddWithValue("@email", p.getEmail());
                    ps.Parameters.AddWithValue("@direccion", p.getDireccion());
                    ps.Parameters.AddWithValue("@activo", p.isActivo());
                    int rows = ps.ExecuteNonQuery();
                    if (rows > 0)
                    {
                        p.setId((int)ps.LastInsertedId);
                        return true;
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return false;
        }

        public bool actualizar(Proveedor p)
        {
            string sql = "UPDATE proveedores SET nombre=@nombre, contacto=@contacto, telefono=@telefono, email=@email, direccion=@direccion WHERE id=@id";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@nombre", p.getNombre());
                    ps.Parameters.AddWithValue("@contacto", p.getContacto());
                    ps.Parameters.AddWithValue("@telefono", p.getTelefono());
                    ps.Parameters.AddWithValue("@email", p.getEmail());
                    ps.Parameters.AddWithValue("@direccion", p.getDireccion());
                    ps.Parameters.AddWithValue("@id", p.getId());
                    return ps.ExecuteNonQuery() > 0;
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return false;
        }

        public bool eliminar(int id)
        {
            string sql = "UPDATE proveedores SET activo = 0 WHERE id = @id";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@id", id);
                    return ps.ExecuteNonQuery() > 0;
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return false;
        }

        private Proveedor mapear(MySqlDataReader rs)
        {
            return new Proveedor(
                rs.GetInt32("id"), 
                rs.GetString("nombre"), 
                rs.GetString("contacto"),
                rs.GetString("telefono"), 
                rs.GetString("email"), 
                rs.GetString("direccion"), 
                rs.GetBoolean("activo")
            );
        }
    }
}

File: ConexionBD/ProveedorBD.py
from ConexionBD.Conexion import Conexion
from Modelo.Proveedor import Proveedor

class ProveedorBD:
    def obtenerTodos(self):
        lista = []
        sql = "SELECT * FROM proveedores WHERE activo = 1 ORDER BY id"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor(dictionary=True)
                cursor.execute(sql)
                rows = cursor.fetchall()
                cursor.close()
                for row in rows:
                    lista.append(self.mapear(row))
        except Exception as e:
            print(e)
        return lista

    def buscarPorId(self, id_prov):
        sql = "SELECT * FROM proveedores WHERE id = %s"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor(dictionary=True)
                cursor.execute(sql, (id_prov,))
                row = cursor.fetchone()
                cursor.close()
                if row:
                    return self.mapear(row)
        except Exception as e:
            print(e)
        return None

    def insertar(self, p):
        sql = "INSERT INTO proveedores (nombre, contacto, telefono, email, direccion, activo) VALUES (%s,%s,%s,%s,%s,%s)"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor()
                cursor.execute(sql, (p.getNombre(), p.getContacto(), p.getTelefono(), p.getEmail(), p.getDireccion(), p.isActivo()))
                conn.commit()
                if cursor.rowcount > 0:
                    p.setId(cursor.lastrowid)
                    cursor.close()
                    return True
                cursor.close()
        except Exception as e:
            print(e)
        return False

    def actualizar(self, p):
        sql = "UPDATE proveedores SET nombre=%s, contacto=%s, telefono=%s, email=%s, direccion=%s WHERE id=%s"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor()
                cursor.execute(sql, (p.getNombre(), p.getContacto(), p.getTelefono(), p.getEmail(), p.getDireccion(), p.getId()))
                conn.commit()
                res = cursor.rowcount > 0
                cursor.close()
                return res
        except Exception as e:
            print(e)
        return False

    def eliminar(self, id_prov):
        sql = "UPDATE proveedores SET activo = 0 WHERE id = %s"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor()
                cursor.execute(sql, (id_prov,))
                conn.commit()
                res = cursor.rowcount > 0
                cursor.close()
                return res
        except Exception as e:
            print(e)
        return False

    def mapear(self, row):
        p = Proveedor()
        p.setId(row["id"])
        p.setNombre(row["nombre"])
        p.setContacto(row["contacto"])
        p.setTelefono(row["telefono"])
        p.setEmail(row["email"])
        p.setDireccion(row["direccion"])
        p.setActivo(bool(row["activo"]))
        return p
